### PR TITLE
Ship tags and descriptions

### PR DESCRIPTION
**Pull request by Latentish**
## About The Pull Request
This adds descriptions and tags to every ship in the game that is not a subshuttle. It also makes these viewable in the Ship Purchase, Ship Join, and Shuttle Manipulator tabs. It also reworks the shuttle manipulator template tab, and adds the ability to add or remove tags from ships manually in the ship editor.
6dd8e53c4b8224837f1d9748c4a37ccf
77def6942b2f4c4fa4424f57b80f9058
cc78d07d7c2e27526ab4c3756dcb7831
eab030d58d6187c8111162a141f761f5

Search features and other ship purchasing tgui related changes will be added in separate PR.

## Why It's Good For The Game
This allows people to get a better sense of what they're buying. It also reworks some years old shuttle manipulator ui.

## Changelog

:cl:phoaly
add: Added descriptions to all jsons
add: Added tags to all ship jsons
add: Added Ship Tags and descriptions to the ship purchase/join/shuttle manipulator menus
add: Added tag manipulation to the ship editor
tweak: Tweaks the UI of the shuttle manipulator
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/ShipSelect.js [hunk 1]
```diff
@@ -101,6 +101,10 @@ export const ShipSelect = (props, context) => {
                 <LabeledList.Item label="Ship Description">
                   {selectedShip.desc || 'No Description'}
                 </LabeledList.Item>
+                <LabeledList.Item label="Ship Tags">
+                  {(selectedShip.tags && selectedShip.tags.join(', ')) ||
+                    'No Tags Set'}
+                </LabeledList.Item>
                 <LabeledList.Item label="Ship Memo">
```
Comment (tmtmtl30) [inline tgui/packages/tgui/interfaces/ShipSelect.js]: you mentioned wanting to move the memo up, right?

File: code/datums/shuttles.dm [hunk 2]
```diff
@@ -194,6 +195,7 @@
 	.["templateName"] = name
 	.["templateShortName"] = short_name
 	.["templateDescription"] = description
+	.["templateTags"] = tags
```
Comment (tmtmtl30) [inline code/datums/shuttles.dm:197]: without modifying the template modification js (ShipEditor.js, i think), this won't do anything

File: code/datums/shuttles.dm [hunk 3]
```diff
@@ -227,6 +229,10 @@
 			description = params["new_template_description"]
 			update_static_data(usr, ui)
 			return TRUE
+		if("setTemplateTags")
```
Comment (tmtmtl30) [inline code/datums/shuttles.dm]: same here -- you'd need to make a button that fires off an act() containing, as a list, the new tags for the template

**Comment by Latentish** [comment]
image
I'll try to fix this tomorrow.

**Comment by Latentish** [comment]
image
This is the only thing I personally touched from the upstream merge. Specifically changing "Info" to "Details".

File: code/datums/shuttles.dm (hunk 2 above)
Comment (Latentish) [reply code/datums/shuttles.dm:197]: Ship editor is fully supported now.

File: code/datums/shuttles.dm (hunk 3 above)
Comment (Latentish) [reply code/datums/shuttles.dm]: Same, it's fully supported now, and this has been split into addTemplateTags and removeTemplateTags.

**Comment by Latentish** [comment]
I intend to add extra features in another pr once this is merged.
Tags being generated as buttons is a top priority for that pr.
Implementing a tag toggle system in the ship editor is pretty low priority (as tag editing is extremely marginal in usecase anyway).
Adding a tag search system in the three places it'd be useful for (shuttle manipulator, ship purchase, and ship join) (High priority).
Adding a link to the webmaps or embedding images of the ships as previews during ship purchase. (on the fence about this)
Adding a way to set up a ship's join settings and memo on purchase (this is gonna be difficult to figure out).

But for now I want to get this merged because these features are tangential to this pr. The sooner this is merged, the less technical debt accrues for ship descriptions and tags needing to be set for new ships. (I'd prefer codeowners set their own descriptions and tags instead of myself.)

**Comment by tmtmtl30** [comment]
LGTM

**Pull request merged**